### PR TITLE
Query: Add TPC test bases

### PR DESCRIPTION
**Pull request by smitpatel**
These only adds disabled test classes

Part of #3170

### CODE REVIEW

File: test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs [hunk 1]
```diff
@@ -10,11 +10,9 @@ public InheritanceContext(DbContextOptions options)
     {
     }
 
-    public DbSet<Animal> Animals { get; set; }
     public DbSet<AnimalQuery> AnimalQueries { get; set; }
     public DbSet<Country> Countries { get; set; }
     public DbSet<Drink> Drinks { get; set; }
-    public DbSet<Plant> Plants { get; set; }
```
Comment (AndriySvyryd) [inline test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs:17]: Why these and `Factions` above were removed?

Comment (smitpatel) [reply test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs:17]: Those are abstract types so convention tried it to map to its table and failed for TPC. Should I file an issue for DbSet to table mapping convention to not do it for abstract type when configured for TPC?

Comment (AndriySvyryd) [reply test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs:17]: Yes please

Comment (smitpatel) [reply test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs:17]: Filed #27947

**Review by AndriySvyryd — APPROVED** [review]

File: test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs (hunk 1 above)
Comment (smitpatel) [reply test/EFCore.Specification.Tests/TestModels/InheritanceModel/InheritanceContext.cs:17]: Added these back and worked around by explicitly mapping to null table for abstract type in TPC.

**Pull request merged**